component/src/main/java/me/androidbox/component/general/PhotoPicker.kt

```diff
@@ -0,0 +1,197 @@
+package me.androidbox.component.general
+
+import android.net.Uri
+import android.util.Log
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.PickVisualMediaRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.BusbyTaskyTheme
+import me.androidbox.component.ui.theme.photoBackgroundColor
+import me.androidbox.component.ui.theme.photoPickerBorderColor
+import me.androidbox.component.ui.theme.photoTextColor
+
+@Composable
+fun PhotoPicker(
+    listOfPhotoUri: SnapshotStateList<Uri>,
+    onPhotoUriSelected: (photoUri: Uri) -> Unit,
+    modifier: Modifier = Modifier) {
+
+    val photoPickerLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.PickVisualMedia(),
+        onResult = { uri ->
+            if (uri != null) {
+                Log.d("PHOTO", "Path of image [ $uri ]")
+                onPhotoUriSelected(uri)
+            }
+        })
+
+    /* We have no images attached */
+    if(listOfPhotoUri.isEmpty()) {
+        AddFirstPhoto(modifier = modifier) {
+            photoPickerLauncher.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly))
```

Comment:
Both the AddFirstPhoto and AddSequentialPhoto have the same callback

Maybe you can find a way to just write it once 
Mostly so in the future, if you want to support other things like video, you just change it in one place